prow/gerrit/adapter/adapter_test.go

```diff
@@ -163,6 +163,66 @@ func fakeProwYAMLGetter(
 	return &res, nil
 }
 
+func TestSkipChangeProcessingChecks(t *testing.T) {
+	now := time.Now()
+	instance := "gke-host"
+	project := "private-cloud"
+	var lastUpdateTime = now.Add(-time.Hour)
+	c := &Controller{
+		configAgent: &config.Agent{},
+	}
+	c.configAgent.Set(&config.Config{ProwConfig: config.ProwConfig{Gerrit: config.Gerrit{AllowedPresubmitTriggerReRawString: ""}}})
+	cases := []struct {
+		name     string
+		instance string
+		change   gerrit.ChangeInfo
+		latest   time.Time
+		result   bool
+	}{
+		{
+			name:     "should not skip change processing when revision is new",
+			instance: instance,
+			change: gerrit.ChangeInfo{ID: "1", CurrentRevision: "10", Project: project,
+				Revisions: map[string]gerrit.RevisionInfo{
+					"10": {Created: makeStamp(now)},
+				}},
+			latest: lastUpdateTime,
+			result: false,
+		},
+		{
+			name:     "should not skip change processing when comment contains test related commands",
+			instance: instance,
+			change: gerrit.ChangeInfo{ID: "1", CurrentRevision: "10", Project: project,
+				Revisions: map[string]gerrit.RevisionInfo{
+					"10": {Number: 10, Created: makeStamp(now.Add(-2 * time.Hour))},
+				}, Messages: []gerrit.ChangeMessageInfo{
+					{
+						Date:           makeStamp(now),
+						Message:        "/test all",
+						RevisionNumber: 10,
+					},
+				}},
+			latest: lastUpdateTime,
+			result: false,
+		},
+		{
+			name:     "should not skip change processing when last project sync time is unknown",
+			instance: instance,
+			change:   gerrit.ChangeInfo{},
+			latest:   time.Time{},
+			result:   false,
```

Comment:
Please add at least one test case where we actually do skip something (i.e. where `result: true`)